### PR TITLE
Add Bulk Actions component

### PR DESCRIPTION
**Pull request by lucasaarcoverde**
#### What is the purpose of this pull request?

* Add Bulk Actions component
* Add `bulk` resolver to the Table

#### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Quality improvement (tests or refactors)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Trivial change (small fix or feature that doesn't impact functionalities)
- [ ] Requires change to documentation, which has been updated accordingly

### CODE REVIEW

File: packages/admin-ui/src/bulk-actions/bulk-actions.state.ts [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+      initialValue: [],
+    })
+
+  const { mapPageItem, ids: pageIds } = useMemo(() => {
+    return pageItems.reduce(
+      (
+        acc: {
+          ids: Array<number | string>
+          mapPageItem: Record<string | number, T>
+        },
```
Comment (matheusps) [inline packages/admin-ui/src/bulk-actions/bulk-actions.state.ts:28]: Create a type for better readability.

File: packages/admin-ui/src/bulk-actions/bulk-actions.state.ts [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+
+    return selectedItemsIds
+  }, [selectedItemsIds])
+
+  const pageSelectedItems = useMemo(
+    () => getSelectedIds().map((item) => mapPageItem[item]),
+    [getSelectedIds, mapPageItem]
+  )
+
+  const mapSelectedIds: Record<string | number, boolean> = useMemo(() => {
```
Comment (matheusps) [inline packages/admin-ui/src/bulk-actions/bulk-actions.state.ts:49]:
```suggestion
  const mapSelectedIds = useMemo<Record<string | number, boolean>>(() => {
```

File: packages/admin-ui/src/table/components/table-body.tsx [hunk 3]
```diff
@@ -79,16 +79,23 @@ export const TableBodyRow = createComponent<'tr', TableBodyRowOptions>(
 
     const isRowSelected = () => {
       const isSelectable = columns.some(
-        (column) => column?.resolver?.type === 'selection'
+        (column) =>
+          column?.resolver?.type === 'selection' ||
+          column?.resolver?.type === 'bulk'
```
Comment (matheusps) [inline packages/admin-ui/src/table/components/table-body.tsx:84]: Maybe that's where we can improve the resolver architecture. Food for thought.

**Review by matheusps — APPROVED** [review]

**Review by anitavincent — APPROVED** [review]

**Review by Lenderson — APPROVED** [review]

File: packages/admin-ui/src/bulk-actions/bulk-actions.i18n.ts [hunk 4]
```diff
@@ -0,0 +1,52 @@
+export const messages = {
+  'en-US': {
+    selectAll: 'Select all',
+    deselectAll: 'Deselect all',
+    selected: '{current} of {total} selected',
+  },
+  'es-AR': {
```
[the remark this answers is not in the document]
Comment (lucasaarcoverde) [reply packages/admin-ui/src/bulk-actions/bulk-actions.i18n.ts:7]: I let them in English and then I open a request on the #product-localization channel to request a review from the localization team

**Pull request merged**